ocaml/xapi/xapi_pool.ml

```diff
@@ -1357,6 +1388,16 @@ let join_force ~__context ~master_address ~master_username ~master_password =
   join_common ~__context ~master_address ~master_username ~master_password
     ~force:true
 
+let pre_join_exchange_certificates ~__context ~uuid ~certificate :
+    API.string_to_string_map =
+  let to_hosts = Xapi_pool_helpers.get_master_slaves_list ~__context in
+  let joiner_certificate =
+    Cert_distrib.make_joiner_cert ~__context ~uuid ~certificate
+  in
+  Cert_distrib.import_joiner ~__context ~joiner_certificate ~to_hosts ;
+  Cert_distrib.make_pool_certs ~__context
+  |> List.map Cert_distrib.host_cert_to_string_to_string
```

Comment:
I think all of this should be internal to `cert_distrib.ml`, then you can get away with exposing a single function in `cert_distrib.mli`. something like:

```
val send_joiner_certs_to_pool : __context -> certificate -> to_hosts -> map
```